lib/rrd.ml

```diff
@@ -445,21 +447,16 @@ let ds_update rrd timestamp values transforms new_domid =
 
 (** Update the rrd with named values rather than just an ordered array *)
 let ds_update_named rrd timestamp ~new_domid valuesandtransforms =
-  let identity x = x in
-  let ds_names = Array.map (fun ds -> ds.ds_name) rrd.rrd_dss in
-  let ds_values =
-    Array.map
-      (fun name ->
-        try fst (List.assoc name valuesandtransforms) with _ -> VT_Unknown
-      )
-      ds_names
+  let valuesandtransforms =
+    valuesandtransforms |> List.to_seq |> StringMap.of_seq
   in
-  let ds_transforms =
-    Array.map
-      (fun name ->
-        try snd (List.assoc name valuesandtransforms) with _ -> identity
-      )
-      ds_names
+  let ds_values, ds_transforms =
+    rrd.rrd_dss
+    |> Array.map (fun ds ->
+           try StringMap.find ds.ds_name valuesandtransforms
```

Comment:
you can also use find_opt here to avoid raising an exception (which requires generating a backtrace and is more costly)